storage/src/test/java/tech/pegasys/teku/storage/client/RecentChainDataTest.java

```diff
@@ -623,10 +636,15 @@ public void getBlockRootBySlot_forOutOfRangeSlot() {
   @Test
   public void getBlockRootBySlot_forHistoricalSlotInRange() {
     initPostGenesis();
-    final UInt64 historicalRoots = UInt64.valueOf(Constants.SLOTS_PER_HISTORICAL_ROOT);
+    final UInt64 historicalRoots = UInt64.valueOf(specConstants.getSlotsPerHistoricalRoot());
     final UInt64 targetSlot = UInt64.valueOf(10);
     final UInt64 finalizedBlockSlot = targetSlot.plus(historicalRoots);
-    final UInt64 finalizedEpoch = compute_epoch_at_slot(finalizedBlockSlot).plus(ONE);
+    final UInt64 finalizedEpoch =
+        specProvider
+            .getGenesisSpec()
```

Comment:
This is kind of annoying but we should probably try to be stricter where possible:
```suggestion
        specProvider
            .atSlot(finalizedBlockSlot)
```